### PR TITLE
feat(sink): add elasticsearch-7 sink and related e2e test

### PR DESCRIPTION
**Pull request by wugouzi**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
Add elasticsearch-7 (version 7.x) remote sink to risingwave. Supported syntax is 
``` sql
CREATE SINK es from test WITH (
    connector = 'elasticsearch-7',
    index = 'es index name',
    url = 'es host'
    username = '<optional>',
    password = '<optional>',
    delimiter = '<optional>'
);
```

Related e2e test is added as well. It checks `hits` field from elasticsearch to verify the correctness.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
~~- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).~~
~~- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).~~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
~~- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)~~

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: e2e_test/sink/elasticsearch/elasticsearch_sink.slt [hunk 1]
```diff
@@ -0,0 +1,33 @@
+statement ok
+CREATE TABLE t7 (v1 int primary key, v2 bigint, v3 varchar);
+
+statement ok
+CREATE MATERIALIZED VIEW mv6 AS SELECT * FROM t7;
```
Comment (tabVersion) [inline e2e_test/sink/elasticsearch/elasticsearch_sink.slt:5]: table contains a mv exec, you can just sink from table here

File: e2e_test/sink/elasticsearch/elasticsearch_sink.slt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+statement ok
+CREATE SINK s7 AS select mv6.v1 as v1, mv6.v2 as v2, mv6.v3 as v3 from mv6 WITH (
+    connector = 'elasticsearch',
+    index = 'test',
+    url = 'http://elasticsearch:9200'
```
Comment (tabVersion) [inline e2e_test/sink/elasticsearch/elasticsearch_sink.slt:11]: will the url contain auth credentials like jdbc? or there are other auth options

File: java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSink7Config.java [hunk 3]
```diff
@@ -30,14 +30,22 @@ public class EsSinkConfig extends CommonSinkConfig {
...
     @JsonCreator
-    public EsSinkConfig(
-            @JsonProperty(value = "url") String url,
-            @JsonProperty(value = "index") String index,
-            @JsonProperty(value = "delimiter") String delimiter) {
+    public EsSink7Config(
+            @JsonProperty(value = "url", required = true) String url,
+            @JsonProperty(value = "index", required = true) String index,
+            @JsonProperty(value = "delimiter") String delimiter,
+            @JsonProperty(value = "username") String username,
```
Comment (wenym1) [inline java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSink7Config.java:42]: Putting the field in the constructor will make the field become required. Is this expected?

**Review by wenym1 — COMMENTED** [review]
Rest LGTM

File: java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSink7Config.java (hunk 3 above)
Comment (wugouzi) [reply java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSink7Config.java:42]: I think this is the requirement for the field in json. url and index are required here. For other sink, the fields is ensured by `mapper.configure(DeserializationFeature.FAIL_ON_MISSING_CREATOR_PROPERTIES, true)`.

Comment (tabVersion) [reply java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSink7Config.java:42]: I believe they are optional fields but e2e also serves as docs in another way. We'd better demonstrate how to connect to a es with auth.

**Review by tabVersion — APPROVED** [review]
basically LGTM, please make username and password optional (or conditional as they must appear together).

File: java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSink7Config.java (hunk 3 above)
Comment (wugouzi) [reply java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSink7Config.java:42]: They are optional. This e2e test connects to es with auth.

Comment (wenym1) [reply java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSink7Config.java:42]: In current code, can users create an es sink without setting the username and password? 

Previously putting a field in the constructor will make the field a required field, no matter whether we mark it with `required = true` or not. So in current code it is likely that user will fail to create the sink without setting the username and password.

Conventionally we don't put the optional field in the constructor and instead mark the optional field with `JsonProperty` annotation.

Comment (wugouzi) [reply java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSink7Config.java:42]: Sorry for the late response. Seems github does not notify me.

Yes, they can. Integration test does this test and at least for now, the missing field will become null. I will check the conventional usage.

**Pull request merged**